### DISCUSSION TITLE
[Feature] PKCE Support?

### DISCUSSION THREAD

**Discussion by KenMacD** [immich-app/immich#4459] [Feature Request]
### The feature

I tried setting up Oauth2 with Kanidm, but I see the PKCE features of the the openid client aren't in use. Any plans to support them?

[The openid-client quickstart](https://www.npmjs.com/package/openid-client#quick-start) includes instructions that might help.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by ToxicMushroom** [comment]
Also see https://kanidm.github.io/kanidm/master/frequently_asked_questions.html#why-is-disabling-pkce-considered-insecure

**Comment by kingp0dd** [comment]
any updates?

**Comment by ToxicMushroom** [comment]
Looks like they added: https://github.com/immich-app/immich/commit/b7a0cf2470cdb4ed1008bd2c13b6e24ceca7d876
Though I didn't see it in the changelog

**Reply by ToxicMushroom** [reply]
I re-enabled pkce for immich in kanidm and it seems to work now :)

**Reply by provokateurin** [reply]
I also tried it and it seems to work, however it is still expected that you provide a client secret which is not necessary for PKCE. For now I just put a dummy value for the secret and there are no issues with doing so it seems. I've opened https://github.com/immich-app/immich/issues/18826 about that.

**Reply by ToxicMushroom** [reply]
I'm not sure I follow. Are you saying Immich became a public client ?

**Reply by provokateurin** [reply]
Yeah so public enforces PKCE and then no client secret is needed.

**Reply by ToxicMushroom** [reply]
I don't see that reflected in the [docs](https://immich.app/docs/administration/oauth/)
I also would love to see the reasoning for making Immich a [public client](https://datatracker.ietf.org/doc/html/rfc6749#section-2.1) ? 

per https://oauth.net/2/pkce/
> PKCE is recommended even if a client is using a client secret or other form of [client authentication](https://oauth.net/2/client-authentication/) like [private_key_jwt](https://oauth.net/private-key-jwt).

Currently what you're describing to me feels like: adding guards (PKCE) next to your locked door (confidential) and the removing the locks (making public) because we have guards now.
